toolkits/golang.go

```diff
@@ -255,48 +255,67 @@ func (toolkit GoToolkit) Install() error {
 
 // === Toolkit: Prepare for Step Run ===
 
-func goBuildInIsolation(packageName, srcPath, outputBinPath string) error {
+func goBuildStep(cmdRunner commandRunner, goConfig GoConfigurationModel, packageName, stepAbsDirPath, outputBinPath string) error {
+	cmdBuilder := goCmdBuilder{goConfig: goConfig}
+
+	if isGoPathModeStep(stepAbsDirPath) {
+		log.Debugf("[Go deps] Step requires GOPATH mode")
+		// Go 1.17 will ignore GO111MODULE (https://blog.golang.org/go116-module-changes)
+		// GO111MODULE needs to be set to "on" when GOPATH is no longer supported.
+		// If GO111MODULE is not set, will be handled as it was "on".
+		mode, err := getGoEnv(cmdRunner, goConfig.GoBinaryPath, "GO111MODULE")
+		if err != nil {
+			log.Warnf("[Go deps] Could not determine if GOPATH mode is supported: %v", err)
+		}
+
+		if isGoPathModeSupported(mode) {
```

Comment:
Are you sure that the `GO111MODULE` will be removed in Go 1.17 and not just ignored?

Reply:
According to https://blog.golang.org/go116-module-changes it will be ignored. We will have to set it to `on`, which is handled the same as the env was removed.

Reply:
Discussed offline, the env will be just ignored, but we have to manually set to `on` on the VMs.